### DISCUSSION TITLE
Integrate Pusher-js with Refine

### DISCUSSION THREAD

**Discussion by awixor** [refinedev/refine#4326] [Q&A] [answered]
Hello,

I am migrating a project from CRA to refine.dev and I wanted to add `pusher-js` for realtime.
I wonder if it's enough to normaly add it like in CTA or should I do it using the `liveProvider`.
What is the differnce between the two methods.

**Comment by alicanerdurmaz** [comment, marked as answer]
Hi @awixor,

When you use `liveProvider`, **refine** becomes aware of it.
For example, defined `resources` can be automatically invalidated.

For more information, visit: https://refine.dev/docs/api-reference/core/providers/live-provider/
